packages/big-design/src/components/Counter/Counter.tsx

```diff
@@ -0,0 +1,214 @@
+import { AddCircleOutlineIcon, RemoveCircleOutlineIcon } from '@bigcommerce/big-design-icons';
+import React, { forwardRef, useMemo, useState, Ref } from 'react';
+
+import { useUniqueId } from '../../hooks';
+import { typedMemo, warning } from '../../utils';
+import { FormControlDescription, FormControlError, FormControlLabel } from '../Form';
+
+import { StyledCounterInput, StyledCounterWrapper } from './styled';
+
+export interface CounterProps extends React.InputHTMLAttributes<HTMLInputElement> {
+  label?: React.ReactChild;
+  labelId?: string;
+  description?: React.ReactChild;
+  error?: React.ReactNode | React.ReactNode[];
+  value: number;
+  step?: number;
+  onCountChange(count: number): void;
+}
+
+interface PrivateProps {
+  forwardedRef: Ref<HTMLInputElement>;
+}
+
+export const StylableCounter: React.FC<CounterProps & PrivateProps> = typedMemo(
+  ({
+    min = 0,
+    max = 100,
+    step = 1,
+    forwardedRef,
+    label,
+    labelId,
+    description,
+    error,
+    disabled,
+    value,
+    onCountChange,
+    ...props
+  }) => {
+    const [focus, setFocus] = useState(false);
+    const uniqueCounterId = useUniqueId('counter');
+    const id = props.id ? props.id : uniqueCounterId;
+
+    const decreaseIconColor = () => {
+      return value <= min ? 'secondary' : undefined;
```

Comment:
One more thing. From the designs, it looks like the base color of the +/- icons are actually `secondary60`.